sa/authz.go

```diff
@@ -0,0 +1,82 @@
+package sa
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	gorp "github.com/letsencrypt/boulder/Godeps/_workspace/src/gopkg.in/gorp.v1"
+
+	"github.com/letsencrypt/boulder/core"
+)
+
+var authorizationTables = []string{
+	"authz",
+	"pendingAuthorizations",
+}
+
+const getAuthorizationIDsMax = 1000
+
+func getAuthorizationIDsByDomain(db *gorp.DbMap, ident string, now time.Time) ([]string, error) {
+	var allIDs []string
+	for _, tableName := range authorizationTables {
+		_, err := db.Select(
+			&allIDs,
+			fmt.Sprintf(
+				`SELECT id FROM %s
+         WHERE identifier = :ident AND
+         status != :invalid AND
+         status != :revoked AND
+         expires > :now
+         LIMIT :limit`,
+				tableName,
+			),
+			map[string]interface{}{
+				"ident":   ident,
+				"invalid": string(core.StatusInvalid),
+				"revoked": string(core.StatusRevoked),
+				"now":     now,
+				"limit":   getAuthorizationIDsMax,
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+		if len(allIDs) == getAuthorizationIDsMax {
```

Comment:
Do we really care about this? If we did, wouldn't we make the limit in the second iteration `getAuthorizationIDsMax - len(allIds)`?

I think we don't and this makes the code clearer.